### PR TITLE
My code review suggestions.

### PR DESCRIPTION
**Pull request by lizbissinger**
Analysis of the program:
The program has some UI set up showing what there intention for the app is. The app still seems to require more backend configuring, but overall the idea is good.

Was the program available in GitHub in time:
Yes

Is the program documented/commented well:
Program has comments when necessary.

Does the program compile:
Yes

Rationale behind my changes:
The changes I made had the intention to clean up and clarify the code.

Links to three specific commits I made to my groups repository:
https://github.com/comstogm/ProjectIT3048C/commit/c64ce77bf38d387fd478783b9c6ef4b1d3ca0253

https://github.com/comstogm/ProjectIT3048C/commit/28d34637c60b7aef09634eec26aba7472516d83a

https://github.com/comstogm/ProjectIT3048C/commit/d166f4d4e432e591411a050d1644d92061de3e3e

Specific technical concepts I learned from this code review:

I learned when to removed redundant code 
I learned when lateinit is necessary/unnecessary
I learned about explicit type arguments

### CODE REVIEW

File: app/src/main/java/com/taskruler/dto/Activity.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ data class Activity (var activityId : Int = 0,
                      var isCompleted : Boolean = false) {
                      //var isCompleted : String = ""
     override fun toString(): String {
-        return "$activityName"
+        return activityName
```
Comment (lizbissinger) [inline app/src/main/java/com/taskruler/dto/Activity.kt:16]: Redundant string template when only single expression
https://kotlinlang.org/docs/basic-syntax.html#ranges

File: app/src/main/java/com/taskruler/MainActivity.kt [hunk 2]
```diff
@@ -121,9 +121,7 @@ fun UserTasksList(
         Button(onClick = {
             selectedUserTask.apply {
                 activityName = inActivityName
-                activityId = selectedActivity?.let {
-                    it.activityId
-                } ?: 0
+                activityId = selectedActivity?.activityId ?: 0
```
Comment (lizbissinger) [inline app/src/main/java/com/taskruler/MainActivity.kt:124]: Redundant 'let' call
https://kotlinlang.org/docs/scope-functions.html#context-object-this-or-it

File: app/src/main/java/com/taskruler/MainActivity.kt [hunk 3]
```diff
@@ -165,7 +163,7 @@ fun UserTasksList(
                 Text(text = userTaskText, fontSize = 18.sp, modifier = Modifier.padding(end = 8.dp))
                 Icon(imageVector = Icons.Filled.ArrowDropDown, contentDescription = "")
                 DropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
-                    userTasks.forEach(){
+                    userTasks.forEach{
```
Comment (lizbissinger) [inline app/src/main/java/com/taskruler/MainActivity.kt:166]: Removed unnecessary parenthesis from function call with lambda
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/for-each.html

File: app/src/main/java/com/taskruler/MainActivity.kt [hunk 4]
```diff
@@ -51,7 +51,7 @@ class MainActivity : ComponentActivity() {
     private var firebaseUser: FirebaseUser? = FirebaseAuth.getInstance().currentUser
     private var inTaskName: String = ""
 
-    private val viewModel: MainViewModel by viewModel<MainViewModel>()
+    private val viewModel: MainViewModel by viewModel()
```
Comment (lizbissinger) [inline app/src/main/java/com/taskruler/MainActivity.kt:54]: Removed explicit type argument

File: app/src/main/java/com/taskruler/MainActivity.kt [hunk 5]
```diff
@@ -82,8 +82,8 @@ class MainActivity : ComponentActivity() {
 @Composable
 fun UserTasksList(
     name: String,
-    activities: List<Activity> = ArrayList<Activity>(),
-    userTasks: List<UserTask> = ArrayList<UserTask>(),
+    activities: List<Activity> = ArrayList(),
```
Comment (lizbissinger) [inline app/src/main/java/com/taskruler/MainActivity.kt:85]: Removed explicit type argument

File: app/src/main/java/com/taskruler/MainActivity.kt [hunk 6]
```diff
@@ -82,8 +82,8 @@ class MainActivity : ComponentActivity() {
...
+    userTasks: List<UserTask> = ArrayList(),
```
Comment (lizbissinger) [inline app/src/main/java/com/taskruler/MainActivity.kt:86]: Removed explicit type argument

File: app/src/main/java/com/taskruler/MainActivity.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/taskruler/MainActivity.kt:124]: Either way works.  
{} is a shave more readable, and allows for future expansion of the function body.

**Comment by discospiff** [comment]
Good suggestions here.  Moderate tech debt reduction.  The group should decide whether or not to harvest these changes.

**Pull request merged**